examples/argo/workflow.yaml

```diff
@@ -0,0 +1,95 @@
+kind: Workflow
+apiVersion: v1
+metadata:
+  generateName: git-ghost-argo-example-
+  labels:
+    app: git-ghost-argo-example
+spec:
+  arguments:
+    parameters:
+    - name: image-prefix
+      globalName: image-prefix
+    - name: image-tag
+      globalName: image-tag
+    - name: git-repo
+      globalName: git-repo
+    - name: git-ghost-repo
+      globalName: git-ghost-repo
+    - name: git-ghost-registry
+      globalName: git-ghost-registry
+    - name: git-commit-hash
+      globalName: git-commit-hash
+    - name: diff-hash
+      globalName: diff-hash
+  entrypoint: steps
+  templates:
+  - name: steps
+    steps:
+    - - name: init
+        template: init
+    - - name: job
+        template: job
+  - name: init
+    initContainers:
+    - name: init
+      image: "{{workflow.parameters.image-prefix}}git-ghost-cli:{{workflow.parameters.image-tag}}"
+      imagePullPolicy: IfNotPresent
+      command: ["bash"]
+      args:
+      - "-c"
+      - "git clone {{workflow.parameters.git-repo}} . && git checkout {{workflow.parameters.git-commit-hash}} && git ghost pull -v {{workflow.parameters.git-commit-hash}} {{workflow.parameters.diff-hash}}"
+      workingDir: /workspace
+      env:
+      - name: GIT_SSH_COMMAND
+        value: "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/git-secret/sshkey"
+      - name: GIT_GHOST_REPO
+        value: "{{workflow.parameters.git-ghost-repo}}"
+      volumeMounts:
```

Comment:
We can remove the `volumeMounts` for init containers by specifying `mirrorVolumeMounts`. However, we have to wait for new release w/ the fix of https://github.com/argoproj/argo/pull/1342.